builtin/credential/jwt/path_role.go

```diff
@@ -561,6 +567,10 @@ func (b *jwtAuthBackend) pathRoleCreateUpdate(ctx context.Context, req *logical.
 		role.ClaimMappings = claimMappings
 	}
 
+	if oauth2Metadata, ok := data.GetOk("oauth2_metadata"); ok {
+		role.Oauth2Metadata = oauth2Metadata.([]string)
```

Comment:
We should probably validate values here and prevent duplicates.

Reply:
The names are validated and duplicates get removed in [path_oidc.go](https://github.com/openbao/openbao/blob/85655b829596ca8d2ab7c0fa33fc4a1b6c5bec1e/builtin/credential/jwt/path_oidc.go#L384) so I think that's fine.  Do you agree?

Reply:
I agree, but doesn't that validate it at the time of use, potentially causing login to fail even though the role config would've been misleadingly successful?

Whereas if we validate it here, we prevent login from failing due to bad values and would fail the role config update, making it clear earlier? 

My 2c!

Reply:
Good point.  I put in an extra check.  Duplicates are not a problem.